### PR TITLE
Add opsets ai.onnx version 20 and ai.onnx.ml version 4 from onnx 1.15

### PR DESCRIPTION
**Pull request by cbourjau**
Add latest opsets and apply updates to existing ones based on onnx 1.15.

# Checklist

- [x] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: tests/type_inference/test_compress.py [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
+import spox.opset.ai.onnx.v18 as op18
+import spox.opset.ai.onnx.v19 as op19
+import spox.opset.ai.onnx.v20 as op20
 from spox._graph import arguments
 from spox._standard import InferenceError
 from spox._type_system import Tensor
 
 
-def test_compress_inference():
+@pytest.mark.parametrize("op", [op17, op18, op19, op20])
```
Comment (jbachurski) [inline tests/type_inference/test_compress.py:12]: Why parameterise these test cases specifically? Compress seems the same between these.

**Review by jbachurski — APPROVED** [review]

File: CHANGELOG.rst [hunk 2]
```diff
@@ -10,6 +10,10 @@ Change log
 0.9.3 (2023-10-23)
 ------------------
 
+**New feature**
```
Comment (cbourjau) [inline CHANGELOG.rst:13]: This belongs to a new release with version `0.10.0`.

File: tests/type_inference/test_compress.py [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
+def test_compress_inference(op):
```
Comment (cbourjau) [inline tests/type_inference/test_compress.py:13]:
```suggestion
def test_compress_inference():
```

File: tests/type_inference/test_compress.py (hunk 1 above)
Comment (cbourjau) [reply tests/type_inference/test_compress.py:12]: Good point!

**Pull request merged**